### ISSUE TITLE
https://html.spec.whatwg.org/multipage/media.html failing to get indexed properly

### ISSUE THREAD

**Issue by sideshowbarker** [yacy/yacy_search_server#119]
My Yacy instance is failing to index https://html.spec.whatwg.org/multipage/media.html properly.

None of the content of that document ends up in the index.

So I’m wondering why it’s not getting indexed and if there’s anything I can do to fix it.

#### More details:

In `/ViewFile.html?url=https%3A%2F%2Fhtml.spec.whatwg.org%2Fmultipage%2Fmedia.html` *Parsed Content*, *Parsed Sentences*, and *Parsed Tokens* just show *“html spec whatwg org multipage media"*

*Original Content from Web* shows that Yacy did fetch the document source successfully.

So given that, I guess the cause is that Yacy’s HTML parser isn’t parsing the content as expected?

There are another ~60 documents from https://html.spec.whatwg.org/multipage/ that Yacy is indexing as expected; https://html.spec.whatwg.org/multipage/parsing.html, for example.

**Comment by sideshowbarker** [comment]
Testing locally with `net.yacy.document.parser.htmlParser`, it seems to get parsed as expected:

```
$ java -cp ./lib/yacycore.jar net.yacy.document.parser.htmlParser \
    https://html.spec.whatwg.org/multipage/media.html
HTML Standard
```

…which is expected title (content of the HTML `title` element from the document source).

So I guess it’s not a parsing failure?

If so then I wonder what other troubleshooting steps I should try in order to isolate what part of the processing chain it is actually failing in.

**Comment by luccioman** [comment]
Hi @sideshowbarker , indeed it is a failure of the HTML parser : it fails with an unexpected NullPointerException, and in that situation the generic parser is applied. That generic parser only extract tokens from the URL, that's why you get the _Parsed Tokens_ "html spec whatwg org multipage media". Unfortunately this time nothing that could help appears in the log, and I had to debug from source to check what happens.

By the way, this issue was already fixed recently (see commit 306a82dd718e03301499e69c48c90e90ba3584ad). So if this really matters for you, please upgrade to latest sources.

**Comment by sideshowbarker** [comment]
@luccioman much thanks—I cloned current master and am now running the instance from that, and it’s indexing the document as expected
